llama_stack/core/server/server.py

```diff
@@ -156,21 +155,28 @@ def translate_exception(exc: Exception) -> HTTPException | RequestValidationErro
         )
 
 
-async def shutdown(app):
-    """Initiate a graceful shutdown of the application.
+class StackApp(FastAPI):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.stack: Stack | None = None
 
-    Handled by the lifespan context manager. The shutdown process involves
-    shutting down all implementations registered in the application.
-    """
-    await shutdown_stack(app.__llama_stack_impls__)
+    def construct_stack(self, config: StackRunConfig):
+        self.stack = Stack(config)
+
+        # This is called from a sync context, so we need to run the initialize() method in a thread pool
```

Comment:
why is a thread pool necessary? you are blocking on the result anyway so why not do `asyncio.run()` in the current context ?